src/Microsoft.Identity.Web/TokenCacheProviders/Session/MsalSessionTokenCacheProvider.cs

```diff
@@ -9,119 +9,119 @@
+		/// <summary>
+		/// Writes the token cache identified by its key to the serialization mechanism.
+		/// </summary>
+		/// <param name="cacheKey">Key for the cache (account ID or app ID).</param>
+		/// <param name="bytes">Blob to write to the cache.</param>
+		/// <returns>A <see cref="Task"/> that completes when a write operation has completed.</returns>
+		protected override async Task WriteCacheBytesAsync(string cacheKey, byte[] bytes)
+		{
+			_sessionLock.EnterWriteLock();
+			try
+			{
+				_logger.LogInformation(string.Format(CultureInfo.InvariantCulture, LogMessages.SerializingSessionCache, _session.Id, cacheKey));
 
-                // Reflect changes in the persistent store
-                _session.Set(cacheKey, bytes);
-                await Task.CompletedTask.ConfigureAwait(false);
-            }
-            finally
-            {
-                _sessionLock.ExitWriteLock();
-            }
-        }
+				// Reflect changes in the persistent store
+				_session.Set(cacheKey, bytes);
+				await Task.CompletedTask.ConfigureAwait(false);
+			}
+			finally
+			{
+				_sessionLock.ExitWriteLock();
+			}
+		}
 
-        /// <summary>
-        /// Removes a cache described by its key.
-        /// </summary>
-        /// <param name="cacheKey">Key of the token cache (user account or app ID).</param>
-        /// <returns>A <see cref="Task"/> that completes when key removal has completed.</returns>
-        protected override async Task RemoveKeyAsync(string cacheKey)
-        {
-            _sessionLock.EnterWriteLock();
-            try
-            {
-                _logger.LogInformation(string.Format(CultureInfo.InvariantCulture, LogMessages.ClearingSessionCache, _session.Id, cacheKey));
+		/// <summary>
+		/// Removes a cache described by its key.
+		/// </summary>
+		/// <param name="cacheKey">Key of the token cache (user account or app ID).</param>
+		/// <returns>A <see cref="Task"/> that completes when key removal has completed.</returns>
+		protected override async Task RemoveKeyAsync(string cacheKey)
+		{
+			_sessionLock.EnterWriteLock();
+			try
+			{
+				_logger.LogInformation(string.Format(CultureInfo.InvariantCulture, LogMessages.ClearingSessionCache, _session.Id, cacheKey));
 
-                // Reflect changes in the persistent store
-                _session.Remove(cacheKey);
-                await Task.CompletedTask.ConfigureAwait(false);
-            }
-            finally
-            {
-                _sessionLock.ExitWriteLock();
-            }
-        }
+				// Reflect changes in the persistent store
+				_session.Remove(cacheKey);
+				await Task.CompletedTask.ConfigureAwait(false);
+			}
+			finally
+			{
+				_sessionLock.ExitWriteLock();
+			}
+		}
 
-        private readonly ReaderWriterLockSlim _sessionLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
-    }
+		private readonly ReaderWriterLockSlim _sessionLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
```

Comment:
ReaderWriterLockSlim is disposable. Thus MsalSessionTokenCacheProvider should also be disposable.